uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -379,9 +397,37 @@ define([
      */
     function initListeners () {
       listeners.push(
-        $rootScope.$on('LeaveRequestPopup::updateBalance', function () {
-          vm.updateBalance();
-          setDaySelectionMode();
+        $rootScope.$on('LeaveRequestPopup::updateAbsenceType', function () {
+          var absenceTypeUnitChanged;
+          var previousAbsenceTypeUnit = vm.selectedAbsenceType.calculation_unit_name;
+
+          vm.selectedAbsenceType = getSelectedAbsenceType();
+          absenceTypeUnitChanged = previousAbsenceTypeUnit !== vm.selectedAbsenceType.calculation_unit_name;
+
+          setOpeningBalance();
+
+          return $q.resolve()
+            .then(absenceTypeUnitChanged && setDaysSelectionMode)
+            .then(function () {
+              if (absenceTypeUnitChanged && isCalculationUnit('hours')) {
+                return loadTimeRangesFromWorkPattern('from');
+              }
+            })
+            .then(setRequestDates)
+            .then(performBalanceChangeCalculation);
+        }),
+        $rootScope.$on('LeaveRequestPopup::updateBalance', function (event, absenceTypesWithBalances) {
+          vm.absenceTypes = absenceTypesWithBalances;
+          vm.selectedAbsenceType = getSelectedAbsenceType();
```

Comment:
why? The `LeaveRequestPopup::updateBalance` event is triggered by the request controller when an absence period has changed, why should we expect that that affects in any way the selected absence type?

Reply:
vm.selectedAbsenceType stores the *balance* as well.